src/Microsoft.XmlSerializer.Generator/src/Microsoft/XmlSerializer/Generator/XmlSerializationReader.cs

```diff
@@ -2344,7 +2353,10 @@ private void WriteMemberElementsIf(Member[] members, Member anyElement, string e
                 if (isSequence)
                     Writer.WriteLine("default:");
                 else
-                    Writer.WriteLine("else {");
+                {
+                    Writer.WriteLine("Finish:");
```

Comment:
Is this the right place for `Finish:`? This seems to be the line of original `else`. I guessed the `Finish` label should be after the original `else` block.

Reply:
You are right. So the Finish tag has to put outside of this method. Seems goto is not the right way in this case and I will switch back to do while(false).